evm/p2p/peer.py

```diff
@@ -168,7 +220,13 @@ def process_msg(self, msg):
         if proto is None:
             raise UnknownProtocolCommand(
                 "No protocol found for cmd_id {}".format(cmd_id))
-        proto.process(cmd_id, msg)
+        decoded = proto.process(cmd_id, msg)
+        if decoded is None:
+            return
+        request_id = decoded.get('request_id')
+        if request_id is not None and request_id in self._pending_replies:
+            callback = self._pending_replies.pop(request_id)
```

Comment:
I think there is a subtle race condition here that may or may not be an issue.

If `request_id` gets `pop`'d between the point where we check it's presence in `self._pending_replies` and this line where we `pop` it, this will fail.  Thoughts on moving this to a `try/except` pattern like the following.

```python
if request_id is not None:
    try:
        cb = self._pending_replies.pop(request_id)
    except KeyError:
        pass
    else:
        callback(decoded)
```

Reply:
Well, that can only happen if we have multiple threads; with asyncio context switches can only happen when you `yield from`. And none of this code is thread-safe at this point. If/when we need multiple threads we'd have to revisit a lot of this stuff to make it thread-safe

Reply:
Thanks for the reminder.  I keep forgetting the async model.